File: Bai_40/main.cpp
#include<bits/stdc++.h>
#include<fstream>
using namespace std;
#define ll long long
string str;
ll n;

ll maxpalin(ll l, ll r)
{
    while (str[l]==str[r]&&l>=0&&r<n)
    {
        l--;
        r++;
    }
    return r - l - 1;
}

int main() 
{
    ifstream ifile("E:/Project Visual Studio Code/Summary_exercises/Bai_40/MAXPALIN.INP");
    ofstream ofile("E:/Project Visual Studio Code/Summary_exercises/Bai_40/MAXPALIN.OUT");

    
    ifile >> n;
    ifile >> str;

    ll ans = 1;
    for (ll i = 1; i < n; i++)
    {
        if (str[i-1]==str[i])
        {
            ans = max(ans, maxpalin(i - 1, i));
        }
    }
    for (ll i = 2; i < n; i++)
    {
        if (str[i-2]==str[i])
        {
            ans = max(ans, maxpalin(i - 2, i));
        }
    }
    ofile << ans;
}

File: Bai_40/main.py
def maxpalin(s, l, r):
    while l >= 0 and r < len(s) and s[l] == s[r]:
        l -= 1
        r += 1
    return r - l - 1

with open("E:/Project Visual Studio Code/Summary_exercises/Bai_40/MAXPALIN.INP", 'r') as ifile:
    n = int(ifile.readline().strip())
    letter = ifile.readline().strip()

    ans = 1
    for i in range(1, n):
        if letter[i - 1] == letter[i]:
            ans = max(ans, maxpalin(letter, i - 1, i))

    for i in range(2, n):
        if letter[i - 2] == letter[i]:
            ans = max(ans, maxpalin(letter, i - 2, i))

with open("E:/Project Visual Studio Code/Summary_exercises/Bai_40/MAXPALIN.OUT", 'w') as ofile:
    ofile.write(str(ans))
